src/stan/variational/advi_params_fullrank.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef STAN__VARIATIONAL__ADVI_PARAMS_FULLRANK__HPP
+#define STAN__VARIATIONAL__ADVI_PARAMS_FULLRANK__HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/fun/LDLT_factor.hpp>
+#include <stan/math/prim/scal/meta/constants.hpp>
+
+#include <stan/math/prim/mat/err/check_square.hpp>
+#include <stan/math/prim/mat/err/check_lower_triangular.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/scal/err/check_not_nan.hpp>
+
+#include <vector>
+
+namespace stan {
+
+  namespace variational {
+
+    class advi_params_fullrank {
+    private:
+      Eigen::VectorXd mu_;      // Mean of location-scale family
+      Eigen::MatrixXd L_chol_;  // Cholesky factor of scale matrix
+                                // NOTE: \Sigma = L_chol_ * L_chol_.transpose()
+      int dimension_;
+
+    public:
+      advi_params_fullrank(const Eigen::VectorXd& mu,
+                           const Eigen::MatrixXd& L_chol) :
+      mu_(mu), L_chol_(L_chol), dimension_(mu.size()) {
+        static const char* function =
+          "stan::variational::advi_params_fullrank";
+
+        stan::math::check_size_match(function,
+                               "Dimension of mean vector",     dimension_,
+                               "Dimension of Cholesky factor", L_chol_.rows() );
+        for (int i = 0; i < dimension_; ++i)
+          stan::math::check_not_nan(function, "Mean vector", mu_(i));
+        stan::math::check_lower_triangular(function,
+                               "Cholesky factor", L_chol_);
+      }
+
+      virtual ~advi_params_fullrank() {}  // No-op
```

Comment:
It's a no-op destructor.  And you don't need
a comment that says "no-op" because it's clear from
the function body being empty.

Also, does it need to be virtual?  Are there extensions
planned for this class?  If not, don't make it virtual.
At which point, it'll be defined implicitly and you don't
need it at all.

- Bob


> On Apr 29, 2015, at 3:38 AM, Rob Trangucci <notifications@github.com> wrote:
> 
> In src/stan/variational/advi_params_fullrank.hpp:
> 
> > +      advi_params_fullrank(const Eigen::VectorXd& mu,
> > +                           const Eigen::MatrixXd& L_chol) :
> > +      mu_(mu), L_chol_(L_chol), dimension_(mu.size()) {
> > +        static const char* function =
> > +          "stan::variational::advi_params_fullrank";
> > +
> > +        stan::math::check_size_match(function,
> > +                               "Dimension of mean vector",     dimension_,
> > +                               "Dimension of Cholesky factor", L_chol_.rows() );
> > +        for (int i = 0; i < dimension_; ++i)
> > +          stan::math::check_not_nan(function, "Mean vector", mu_(i));
> > +        stan::math::check_lower_triangular(function,
> > +                               "Cholesky factor", L_chol_);
> > +      }
> > +
> > +      virtual ~advi_params_fullrank() {}  // No-op
> 
> @akucukelbir Do you need this no-op constructor?
> 
> —
> Reply to this email directly or view it on GitHub.
>